internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -26,22 +26,26 @@ const (
 	appTagKey     = "ecs-application"
 )
 
-// CloudFormation wraps the CloudFormationAPI interface
-type CloudFormation struct {
-	client cloudformationiface.CloudFormationAPI
-	box    packd.Box
-}
-
 type stackConfiguration interface {
 	StackName() string
 	Template() (string, error)
 	Parameters() []*cloudformation.Parameter
 	Tags() []*cloudformation.Tag
 }
 
+// CloudFormation wraps the CloudFormationAPI interface
+type CloudFormation struct {
+	regionalClientProvider func(region string) cloudformationiface.CloudFormationAPI
+	client                 cloudformationiface.CloudFormationAPI
+	box                    packd.Box
```

Comment:
The reason we create them here is because they're actually a little expensive. Is there something that concerns you about creating the client like this? Is it that for you, you don't need the box?